os/kernel/log_dump/log_dump.c

```diff
@@ -579,7 +579,7 @@ int log_dump(int argc, char *argv[])
 		} else {
 			/* compress the completely filled block */
 			writesize = CONFIG_LOG_DUMP_CHUNK_SIZE;
-			compress_ret = compress_block(out_buf, &writesize, uncomp_buf[comp_idx], CONFIG_LOG_DUMP_CHUNK_SIZE);
+			compress_ret = compress_block(out_buf,  (long unsigned int *)&writesize,  (unsigned char *)uncomp_buf[comp_idx], CONFIG_LOG_DUMP_CHUNK_SIZE);
```

Comment:
If we change the type of `uncomp_buf` to `unsigned char *`, what happened?

Reply:
I will change and check it out